# Issue 26770: Patch pynac with PR: 336

archive/issues_026533.json:
```json
{
    "body": "Following #25979, this patch apply the modification from E.Bray's [PR:336](https://github.com/pynac/pynac/pull/336).\n\nThis fix bugs in python3 like this one :\n\n```\nsage: log(int(88705158), 4)\n1\n```\n\n**CC:**  @fchapoton @embray @rwst\n\n**Keywords:** pynac, python3\n\n**Branch:** [u/vklein/patch_pynac_with_pr__366](https://github.com/sagemath/sagetrac-mirror/tree/u/vklein/patch_pynac_with_pr__366)\n\n**Commit:** [3907c0bb31edbea6ebb2f03397d731d0f00c80c4](https://github.com/sagemath/sagetrac-mirror/commit/3907c0bb31edbea6ebb2f03397d731d0f00c80c4)\n\n**Author:** Vincent Klein\n\nIssue created by migration from https://trac.sagemath.org/ticket/26770\n\n",
    "closed_at": "2018-12-11T11:11:28Z",
    "created_at": "2018-11-26T15:44:45Z",
    "labels": [
        "component: packages: standard",
        "bug",
        "wontfix"
    ],
    "title": "Patch pynac with PR: 336",
    "type": "issue",
    "url": "https://github.com/sagemath/sage/issues/26770",
    "user": "https://github.com/vinklein"
}
```
Following #25979, this patch apply the modification from E.Bray's [PR:336](https://github.com/pynac/pynac/pull/336).

This fix bugs in python3 like this one :

```
sage: log(int(88705158), 4)
1
```

**CC:**  @fchapoton @embray @rwst

**Keywords:** pynac, python3

**Branch:** [u/vklein/patch_pynac_with_pr__366](https://github.com/sagemath/sagetrac-mirror/tree/u/vklein/patch_pynac_with_pr__366)

**Commit:** [3907c0bb31edbea6ebb2f03397d731d0f00c80c4](https://github.com/sagemath/sagetrac-mirror/commit/3907c0bb31edbea6ebb2f03397d731d0f00c80c4)

**Author:** Vincent Klein

Issue created by migration from https://trac.sagemath.org/ticket/26770





---

archive/issue_comments_414892.json:
```json
{
    "body": "**Branch:** [u/vklein/patch_pynac_with_pr__366](https://github.com/sagemath/sagetrac-mirror/tree/u/vklein/patch_pynac_with_pr__366)",
    "created_at": "2018-11-26T15:49:05Z",
    "issue": "https://github.com/sagemath/sage/issues/26770",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/26770#issuecomment-414892",
    "user": "https://github.com/vinklein"
}
```

**Branch:** [u/vklein/patch_pynac_with_pr__366](https://github.com/sagemath/sagetrac-mirror/tree/u/vklein/patch_pynac_with_pr__366)



---

archive/issue_comments_414893.json:
```json
{
    "body": "**Author:** Vincent Klein",
    "created_at": "2018-11-26T15:51:11Z",
    "issue": "https://github.com/sagemath/sage/issues/26770",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/26770#issuecomment-414893",
    "user": "https://github.com/vinklein"
}
```

**Author:** Vincent Klein



---

archive/issue_comments_414894.json:
```json
{
    "body": "**Commit:** [3907c0bb31edbea6ebb2f03397d731d0f00c80c4](https://github.com/sagemath/sagetrac-mirror/commit/3907c0bb31edbea6ebb2f03397d731d0f00c80c4)",
    "created_at": "2018-11-26T15:51:11Z",
    "issue": "https://github.com/sagemath/sage/issues/26770",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/26770#issuecomment-414894",
    "user": "https://github.com/vinklein"
}
```

**Commit:** [3907c0bb31edbea6ebb2f03397d731d0f00c80c4](https://github.com/sagemath/sagetrac-mirror/commit/3907c0bb31edbea6ebb2f03397d731d0f00c80c4)



---

archive/issue_events_238519.json:
```json
{
    "actor": "https://github.com/vinklein",
    "created_at": "2018-11-26T15:51:11Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/26770",
    "label": "needs review",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/26770#event-238519"
}
```



---

archive/issue_comments_414895.json:
```json
{
    "body": "<a id='comment:2'></a>\n**New commits:**\n<table><tr><td><a href=\"https://github.com/sagemath/sagetrac-mirror/commit/3907c0bb31edbea6ebb2f03397d731d0f00c80c4\">3907c0b</a></td><td><code>Trac #26770: Patch pynac to fix bug with long base ...</code></td></tr></table>\n",
    "created_at": "2018-11-26T15:51:11Z",
    "issue": "https://github.com/sagemath/sage/issues/26770",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/26770#issuecomment-414895",
    "user": "https://github.com/vinklein"
}
```

<a id='comment:2'></a>
**New commits:**
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/3907c0bb31edbea6ebb2f03397d731d0f00c80c4">3907c0b</a></td><td><code>Trac #26770: Patch pynac to fix bug with long base ...</code></td></tr></table>




---

archive/issue_comments_414896.json:
```json
{
    "body": "**Changing keywords** from \"\" to \"pynac, python3\".",
    "created_at": "2018-11-26T16:02:34Z",
    "issue": "https://github.com/sagemath/sage/issues/26770",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/26770#issuecomment-414896",
    "user": "https://github.com/vinklein"
}
```

**Changing keywords** from "" to "pynac, python3".



---

archive/issue_comments_414897.json:
```json
{
    "body": "<a id='comment:4'></a>\nWouldn't it be better to just cut a new pynac bugfix release? Since it advertises itself as a \"Sage Math support library\" it  doesn't make much sense that sage would have to patch it.",
    "created_at": "2018-11-26T19:23:38Z",
    "issue": "https://github.com/sagemath/sage/issues/26770",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/26770#issuecomment-414897",
    "user": "https://github.com/timokau"
}
```

<a id='comment:4'></a>
Wouldn't it be better to just cut a new pynac bugfix release? Since it advertises itself as a "Sage Math support library" it  doesn't make much sense that sage would have to patch it.



---

archive/issue_comments_414898.json:
```json
{
    "body": "<a id='comment:6'></a>\nFor some reason I thought this was already fixed--indeed the solution here would be to release a new pynac.",
    "created_at": "2018-11-27T10:02:41Z",
    "issue": "https://github.com/sagemath/sage/issues/26770",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/26770#issuecomment-414898",
    "user": "https://github.com/embray"
}
```

<a id='comment:6'></a>
For some reason I thought this was already fixed--indeed the solution here would be to release a new pynac.



---

archive/issue_comments_414899.json:
```json
{
    "body": "<a id='comment:7'></a>\nThis is the best solution of course. My only question is when can we expect a new release?",
    "created_at": "2018-11-27T10:52:16Z",
    "issue": "https://github.com/sagemath/sage/issues/26770",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/26770#issuecomment-414899",
    "user": "https://github.com/vinklein"
}
```

<a id='comment:7'></a>
This is the best solution of course. My only question is when can we expect a new release?



---

archive/issue_comments_414900.json:
```json
{
    "body": "**Description changed:**\n``````diff\n--- \n+++ \n@@ -1,4 +1,4 @@\n-Following #25979, this patch apply the modification from E.Bray's [PR:366](https://github.com/pynac/pynac/pull/336).\n+Following #25979, this patch apply the modification from E.Bray's [PR:336](https://github.com/pynac/pynac/pull/336).\n \n This fix bugs in python3 like this one :\n \n``````\n",
    "created_at": "2018-12-08T19:18:04Z",
    "issue": "https://github.com/sagemath/sage/issues/26770",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/26770#issuecomment-414900",
    "user": "https://github.com/fchapoton"
}
```

**Description changed:**
``````diff
--- 
+++ 
@@ -1,4 +1,4 @@
-Following #25979, this patch apply the modification from E.Bray's [PR:366](https://github.com/pynac/pynac/pull/336).
+Following #25979, this patch apply the modification from E.Bray's [PR:336](https://github.com/pynac/pynac/pull/336).
 
 This fix bugs in python3 like this one :
 
``````




---

archive/issue_events_238520.json:
```json
{
    "actor": "https://github.com/fchapoton",
    "created_at": "2018-12-08T19:18:04Z",
    "event": "renamed",
    "issue": "https://github.com/sagemath/sage/issues/26770",
    "rename": {
        "from": "Patch pynac with PR: 366",
        "to": "Patch pynac with PR: 336"
    },
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/26770#event-238520"
}
```



---

archive/issue_comments_414901.json:
```json
{
    "body": "<a id='comment:9'></a>\nPatching packages from within Sage is not a bad thing. I just hope my upcoming Pynac upgrade ticket gets more review than this one.",
    "created_at": "2018-12-10T14:15:55Z",
    "issue": "https://github.com/sagemath/sage/issues/26770",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/26770#issuecomment-414901",
    "user": "https://github.com/rwst"
}
```

<a id='comment:9'></a>
Patching packages from within Sage is not a bad thing. I just hope my upcoming Pynac upgrade ticket gets more review than this one.



---

archive/issue_comments_414902.json:
```json
{
    "body": "<a id='comment:10'></a>\nI disagree, patching packages from within sage *is* a bad thing. It greatly increases the workload for packagers, breaks assumptions of users and may result in confusing upstream bug reports. So at least from my point of view this did get reviewed, with the verdict being \"lets just do a proper update instead\".",
    "created_at": "2018-12-10T18:45:04Z",
    "issue": "https://github.com/sagemath/sage/issues/26770",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/26770#issuecomment-414902",
    "user": "https://github.com/timokau"
}
```

<a id='comment:10'></a>
I disagree, patching packages from within sage *is* a bad thing. It greatly increases the workload for packagers, breaks assumptions of users and may result in confusing upstream bug reports. So at least from my point of view this did get reviewed, with the verdict being "lets just do a proper update instead".



---

archive/issue_comments_414903.json:
```json
{
    "body": "<a id='comment:11'></a>\nReplying to [@timokau](#comment%3A10):\n> I disagree, patching packages from within sage *is* a bad thing.\n\n+1 we're trying to reduce the dependence on \"sage-the-distribution\" which has created an inordinate maintenance burden for the entire community.",
    "created_at": "2018-12-11T10:18:46Z",
    "issue": "https://github.com/sagemath/sage/issues/26770",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/26770#issuecomment-414903",
    "user": "https://github.com/embray"
}
```

<a id='comment:11'></a>
Replying to [@timokau](#comment%3A10):
> I disagree, patching packages from within sage *is* a bad thing.

+1 we're trying to reduce the dependence on "sage-the-distribution" which has created an inordinate maintenance burden for the entire community.



---

archive/issue_events_238521.json:
```json
{
    "actor": "https://github.com/vinklein",
    "created_at": "2018-12-11T10:27:25Z",
    "event": "demilestoned",
    "issue": "https://github.com/sagemath/sage/issues/26770",
    "milestone": "sage-8.5",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/26770#event-238521"
}
```



---

archive/issue_comments_414904.json:
```json
{
    "body": "<a id='comment:12'></a>\nOk then. Wontfix seems to be the majority view for this ticket",
    "created_at": "2018-12-11T10:27:25Z",
    "issue": "https://github.com/sagemath/sage/issues/26770",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/26770#issuecomment-414904",
    "user": "https://github.com/vinklein"
}
```

<a id='comment:12'></a>
Ok then. Wontfix seems to be the majority view for this ticket



---

archive/issue_events_238522.json:
```json
{
    "actor": "https://github.com/vinklein",
    "created_at": "2018-12-11T10:27:45Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/26770",
    "label": "needs review",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/26770#event-238522"
}
```



---

archive/issue_events_238523.json:
```json
{
    "actor": "https://github.com/vinklein",
    "created_at": "2018-12-11T10:27:45Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/26770",
    "label": "positive review",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/26770#event-238523"
}
```



---

archive/issue_events_238524.json:
```json
{
    "actor": "https://github.com/embray",
    "created_at": "2018-12-11T11:11:28Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/26770",
    "label": "positive review",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/26770#event-238524"
}
```



---

archive/issue_events_238525.json:
```json
{
    "actor": "https://github.com/embray",
    "created_at": "2018-12-11T11:11:28Z",
    "event": "closed",
    "issue": "https://github.com/sagemath/sage/issues/26770",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/26770#event-238525"
}
```



---

archive/issue_events_238526.json:
```json
{
    "actor": "https://github.com/embray",
    "created_at": "2018-12-11T11:11:28Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/26770",
    "label": "wontfix",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/26770#event-238526"
}
```
